tests/TestCase/Controller/Component/SecurityComponentTest.php

```diff
@@ -823,11 +906,67 @@ public function testValidatePostFailNoDisabled()
             '_Token' => compact('fields')
         ];
 
-        $result = $this->Controller->Security->validatePost($this->Controller);
+        $result = $this->validatePost('SecurityException', '\'_Token.unlocked\' was not found in request data.');
         $this->assertFalse($result);
     }
 
     /**
+     * test that missing 'debug' input causes failure
+     *
+     * @return void
+     * @triggers Controller.startup $this->Controller
+     */
+    public function testValidatePostFailNoDebug()
+    {
+        $event = new Event('Controller.startup', $this->Controller);
+        $this->Controller->Security->startup($event);
+        $fields = ['Model.hidden', 'Model.password', 'Model.username'];
+        $fields = urlencode(Security::hash(serialize($fields) . Security::salt()));
+        $unlocked = '';
+
+        $this->Controller->request->data = [
+            'Model' => [
+                'username' => 'mark',
+                'password' => 'sekret',
+                'hidden' => '0'
+            ],
+            '_Token' => compact('fields', 'unlocked')
+        ];
+
+        $result = $this->validatePost('SecurityException', '\'_Token.debug\' was not found in request data.');
+        $this->assertFalse($result);
+    }
+
+    /**
+     * test that missing 'debug' input is not the problem when debug mode disabled
+     *
+     * @return void
+     * @triggers Controller.startup $this->Controller
+     */
+    public function testValidatePostFailNoDebugMode()
+    {
+        $event = new Event('Controller.startup', $this->Controller);
+        $this->Controller->Security->startup($event);
+        $fields = ['Model.hidden', 'Model.password', 'Model.username'];
+        $fields = urlencode(Security::hash(serialize($fields) . Security::salt()));
+        $unlocked = '';
+
+        $this->Controller->request->data = [
+            'Model' => [
+                'username' => 'mark',
+                'password' => 'sekret',
+                'hidden' => '0'
+            ],
+            '_Token' => compact('fields', 'unlocked')
+        ];
+        $debug = Configure::read('debug');
+        Configure::write('debug', false);
+        $result = $this->validatePost();
+        $this->assertFalse($result);
+        Configure::write('debug', $debug);
```

Comment:
You shouldn't need to restore. the test harness resets Configure data on each test method.